integrationTests/multiShard/endOfEpoch/rating/rating_test.go

```diff
@@ -0,0 +1,656 @@
+package rating
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	"github.com/ElrondNetwork/elrond-go/integrationTests"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/mock"
+	"github.com/ElrondNetwork/elrond-go/process/rating"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/storage/lrucache"
+)
+
+type testData struct {
+	consensusSize     int
+	shardSize         int
+	metaConsensusSize int
+	metaShardSize     int
+	waitingSize       int
+	offlineNodes      int
+	epochs            int
+	roundDuration     int
+	roundsPerEpoch    int
+	numShards         int
+}
+
+func TestComputeRating_Metachain(t *testing.T) {
+	t.Skip("this should be only activated for computing the outputs")
+
+	datas := createTestData()
+
+	for _, testData := range datas {
+		computeRatings(testData)
+	}
+}
+
+func computeRatings(testData testData) {
+	nodesPerShard := testData.shardSize
```

Comment:
make some smaller functions

Reply:
Agree, the whole file can be split in multiple functions, like steps.

Reply:
fixed